src/app/repositories/firebase/dataset_firebase_repository.py

```diff
@@ -138,6 +147,21 @@ def delete_sub_collection_in_batches(
             logger.error(f"Error deleting sub collection in batches: {e}")
             raise RuntimeError("Error deleting sub collection in batches.")
 
+
+    def get_serialized_size(self, obj) -> int:
+        """
+        Calculate the size of serialized object in bytes.
+
+        Parameters:
+        obj (Any): The object to be serilaized and measured
+
+        Returns:
+        int: The size of the serialized object in bytes
+        """
+        serialized_obj = json.dumps(obj)
+        return len(serialized_obj.encode('utf-8'))
```

Comment:
Move this function to shared_service as it is not part of database interaction.
Also, write unit test for this function.